tests/regression/copy_stencil_tuple.cpp

```diff
@@ -0,0 +1,208 @@
+            };
+        comp();
+        TypeParam::verify(in, out);
+        TypeParam::benchmark("copy_stencil_tuple_extra_dim", comp);
+    }
+
+    GT_REGRESSION_TEST(copy_stencil_tuple_composite, test_environment<>, stencil_backend_t) {
+        auto in = [](int i, int j, int k) { return i + j + k; };
+        auto out0 = TypeParam::make_storage();
+        auto out1 = TypeParam::make_storage();
+        auto out2 = TypeParam::make_storage();
+        auto out3 = TypeParam::make_storage();
+        auto out = sid::composite::keys<integral_constant<int, 0>,
+            integral_constant<int, 1>,
+            integral_constant<int, 2>,
+            integral_constant<int, 3>>::make_values(out0, out1, out2, out3);
+        auto in0 = TypeParam::make_const_storage(in);
+        auto in1 = TypeParam::make_const_storage(in);
+        auto in2 = TypeParam::make_const_storage(in);
+        auto in3 = TypeParam::make_const_storage(in);
+        auto in_sid = sid::composite::keys<integral_constant<int, 0>,
+            integral_constant<int, 1>,
+            integral_constant<int, 2>,
+            integral_constant<int, 3>>::make_values(in0, in1, in2, in3);
+        auto comp = [&out, grid = TypeParam::make_grid(), &in = in_sid] {
+            run_single_stage(copy_functor_tuple(), stencil_backend_t(), grid, in, out);
+        };
+        comp();
+        TypeParam::verify(in, out0);
+        TypeParam::verify(in, out1);
+        TypeParam::verify(in, out2);
+        TypeParam::verify(in, out3);
+        TypeParam::benchmark("copy_stencil_tuple_composite", comp);
+    }
+
+    GT_REGRESSION_TEST(copy_stencil_tuple_dim2tuple, test_environment<>, stencil_backend_t) {
+        using float_t = typename TypeParam::float_t;
+        auto in = [](int i, int j, int k, int t) { return i + j + k + t; };
+        auto out_ds = TypeParam::template builder<float_t>(integral_constant<int, 4>{}).build();
+        auto out = sid::dimension_to_tuple_like<integral_constant<int, 3>, 4>(out_ds);
+        auto in_ds = TypeParam::template builder<float_t const>(integral_constant<int, 4>{}).initializer(in).build();
+        auto in_transformed = sid::dimension_to_tuple_like<integral_constant<int, 3>, 4>(in_ds);
+        auto comp = [&out, grid = TypeParam::make_grid(), &in = in_transformed] {
+            run_single_stage(copy_functor_tuple(), stencil_backend_t(), grid, in, out);
+        };
+        comp();
+        TypeParam::verify(in, out_ds);
+        TypeParam::benchmark("copy_stencil_tuple_dim2tuple", comp);
+    }
+
+    GT_REGRESSION_TEST(copy_stencil_tuple_dim2tuple_unrolled, test_environment<>, stencil_backend_t) {
+        using float_t = typename TypeParam::float_t;
+        auto in = [](int i, int j, int k, int t) { return i + j + k + t; };
+        auto out_ds = TypeParam::template builder<float_t>(integral_constant<int, 4>{}).build();
+        static_assert(is_sid<decltype(out_ds)>::value);
```

Comment:
This line is basically testing test code, so should probably not be here, but in the builder?

Reply:
was debugging something... deleted